server/src/main/java/dev/slimevr/status/StatusSystem.kt

```diff
@@ -0,0 +1,66 @@
+package dev.slimevr.status
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet
+import solarxr_protocol.rpc.StatusDataUnion
+import solarxr_protocol.rpc.StatusMessageT
+import java.util.concurrent.CopyOnWriteArrayList
+import java.util.concurrent.atomic.AtomicInteger
+
+class StatusSystem {
+	private val listeners: MutableList<StatusListener> = CopyOnWriteArrayList()
+	private val statuses: MutableMap<Int, StatusDataUnion> = Int2ObjectOpenHashMap()
+	private val prioritizedStatuses: MutableSet<Int> = IntOpenHashSet()
+	private val idCounter = AtomicInteger(1)
+
+	fun addListener(listener: StatusListener) {
+		listeners.add(listener)
+	}
+
+	fun removeListener(listener: StatusListener) {
+		listeners.remove(listener)
+	}
+
+	fun getStatuses(): Array<StatusMessageT> = statuses.map { (id, message) ->
+		val status = StatusMessageT()
+		status.id = id.toUInt().toLong()
```

Comment:
I don't quite get this lol
```suggestion
		status.id = id.toLong()
```

Reply:
``-52.toLong()`` would just be ``-52``, ``-52.toUInt()`` would convert it into ``4294967244``

Reply:
Why would you not want it to go in the negatives? For solarxr?

Reply:
We are in Kotlin, no purpose of being forced to use `int`s. But I'm still kind of forced to >~>